### ISSUE TITLE
Pushing model to Hugging Face Hub failure

### ISSUE THREAD

**Issue by wmailn** [gensyn-ai/rl-swarm#86]
{'loss': 0.0, 'grad_norm': 5.3411760330200195, 'learning_rate': 0.0, 'rewards/consensus_reward_func': 0.1875, 'rewards/concensus_correctness_reward_func': 0.11949999630451202, 'rewards/question_recreation_reward_func': 0.5229332270100713, 'rewards/final_correctness_reward_func': 0.4375, 'rewards/strict_format_reward_func': 0.0, 'rewards/soft_format_reward_func': 0.0, 'rewards/xmlcount_reward_func': 0.48656249325722456, 'rewards/cumulative_reward_2': 0.0, 'reward': 1.7539957100525498, 'reward_std': 0.7729681366472505, 'completion_length': 376.6875, 'kl': 0.00522619261755608, 'epoch': 1.0}
100%|██████████████████████████████████████████████████████████████████████████████████████████████                                                                                               {'train_runtime': 351.803, 'train_samples_per_second': 0.91, 'train_steps_per_second': 0.057, 'train_loss': 2.8522685170173644e-06, 'epoch': 1.0}                                                 
100%|██████████████████████████████████████████████████████████████████████████████████████████████                                                                                               
100%|██████████████████████████████████████████████████████████████████████████████████████████████                                                                                               ███████████████████████████████████████████████████████████████| 20/20 [05:51<00:00, 17.59s/it]
***** train metrics *****
  total_flos               =        0GF
  train_loss               =        0.0
  train_runtime            = 0:05:51.80
  train_samples            =        160
  train_samples_per_second =       0.91
  train_steps_per_second   =      0.057
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:strong wise gecko:Saving model
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:strong wise gecko:Model saved to runs/gsm8k/multinode/Qwen2.5-0.5B-Instruct-Gensyn-Swarm-strong_wise_gecko
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:strong wise gecko:Tokenizer saved to runs/gsm8k/multinode/Qwen2.5-0.5B-Instruct-Gensyn-Swarm-strong_wise_gecko
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:strong wise gecko:📉 Finished training round: 98 stage: 2
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:strong wise gecko:Pushing model to Hugging Face Hub...
model.safetensors:   0%|                                                                                                                                              | 0.00/1.98G [10:28<?, ?B/s]
  0%|                                                                                                                                                                       | 0/1 [10:28<?, ?it/s]
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:strong wise gecko:Failed to push model to the Hugging Face Hub. When you conclude training please try manually pushing it yourself using the instructions here: https://huggingface.co/docs/hub/en/models-uploading
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:strong wise gecko:🏆 Submitting winners for round 98: ['Qma6orUHgCmEdaaKouySHEutupKcgUv1uSVqn56DZ3zXma']
WARN:urllib3.connectionpool:Connection pool is full, discarding connection: stats.vllm.ai. Connection pool size: 10
WARN:urllib3.connectionpool:Connection pool is full, discarding connection: stats.vllm.ai. Connection pool size: 10
WARN:urllib3.connectionpool:Connection pool is full, discarding connection: stats.vllm.ai. Connection pool size: 10
WARN:urllib3.connectionpool:Connection pool is full, discarding connection: stats.vllm.ai. Connection pool size: 10
WARN:urllib3.connectionpool:Connection pool is full, discarding connection: stats.vllm.ai. Connection pool size: 10
WARN:urllib3.connectionpool:Connection pool is full, discarding connection: stats.vllm.ai. Connection pool size: 10
WARN:urllib3.connectionpool:Connection pool is full, discarding connection: stats.vllm.ai. Connection pool size: 10
WARN:urllib3.connectionpool:Connection pool is full, discarding connection: stats.vllm.ai. Connection pool size: 10
WARN:urllib3.connectionpool:Connection pool is full, discarding connection: stats.vllm.ai. Connection pool size: 10
WARN:urllib3.connectionpool:Connection pool is full, discarding connection: stats.vllm.ai. Connection pool size: 10
WARN:urllib3.connectionpool:Connection pool is full, discarding connection: stats.vllm.ai. Connection pool size: 10
WARN:urllib3.connectionpool:Connection pool is full, discarding connection: stats.vllm.ai. Connection pool size: 10
WARN:urllib3.connectionpool:Connection pool is full, discarding connection: stats.vllm.ai. Connection pool size: 10
WARN:urllib3.connectionpool:Connection pool is full, discarding connection: stats.vllm.ai. Connection pool size: 10
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:strong wise gecko:Already finished round: 98. Next check in 5.0s.
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:strong wise gecko:Already finished round: 98. Next check in 10.0s.
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:strong wise gecko:Already finished round: 98. Next check in 20.0s.
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:strong wise gecko:Already finished round: 98. Next check in 40.0s.
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:strong wise gecko:Already finished round: 98. Next check in 80.0s.
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:strong wise gecko:Already finished round: 98. Next check in 160.0s.
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:strong wise gecko:Already finished round: 98. Next check in 300.0s.

**Comment by h-grieve** [comment]
Did you register a HF key during setup?

**Comment by wmailn** [comment]
> Did you register a HF key during setup?

Yes, I registered and set up my HF key

**Comment by h-grieve** [comment]
Thanks for this. Tagging in @gpandrade

**Comment by wmailn** [comment]
> Thanks for this. Tagging in [@gpandrade](https://github.com/gpandrade)

Is it too network-hugging to upload a 1.98GB file to Hugging Face Hub after each training session? Will this problem be optimized in future development?

**Comment by vijay1528** [comment]
have you used this command (git config --global credential.helper store) before starting the node

**Comment by h-grieve** [comment]
This issue hasn't reoccured so closing ticket for now. Please re-open if you see it again.
